include/picongpu/particles/startPosition/generic/FreeRng.hpp

```diff
@@ -68,9 +68,10 @@ namespace picongpu
                         }
 
                         template<typename T_Particle>
-                        HDINLINE uint32_t numberOfMacroParticles(float_X const realParticlesPerCell)
+                        HDINLINE uint32_t
+                        numberOfMacroParticles(float_X const realParticlesPerCell, DataSpace<simDim> const cellIdx)
                         {
-                            return Functor::template numberOfMacroParticles<T_Particle>(realParticlesPerCell);
+                            return Functor::template numberOfMacroParticles<T_Particle>(realParticlesPerCell, cellIdx);
```

Comment:
This interface change would require an update of all free functors in our examples.
Nevertheless, it introduces the mixing of the separation of density profile and in-cell distribution.

In general, such a change if needed requires a deep review of `ParticlesInit.kernel` to avoid site effects due to the mixed concepts.